orion/primitives/aer.py

```diff
@@ -0,0 +1,316 @@
+        Args:
+            X (ndarray):
+                N-dimensional array containing the input sequences for the model.
+            y (ndarray):
+                N-dimensional array containing the output sequences we want to reconstruct.
+        """
+        if y is None:
+            y = X.copy()  # Reconstruct the same input.
+
+        X = X[:, 1:-1, :]
+        ry, y, fy = y[:, 0], y[:, 1:-1], y[:, -1]
+
+        if not self._fitted:
+            self._augment_hyperparameters(X, y, kwargs)
+            self._build_aer(**kwargs)
+
+        callbacks = [
+            callback['class'](**callback.get('args', dict()))
+            for callback in self.callbacks
+        ]
+
+        self.fit_history = self.aer_model.fit(
+            X, [ry, y, fy], batch_size=self.batch_size, epochs=self.epochs,
+            shuffle=self.shuffle, verbose=self.verbose, callbacks=callbacks,
+            validation_split=self.validation_split
+        )
+
+        self._fitted = True
+
+    def predict(self, X: np.ndarray) -> tuple:
+        """Predict values using the initialized object.
+
+        Args:
+            X (ndarray):
+                N-dimensional array containing the input sequences for the model.
+
+        Returns:
+            ndarray:
+                N-dimensional array containing the regression for each input sequence (forward).
+            ndarray:
+                N-dimensional array containing the reconstructions for each input sequence.
+            ndarray:
+                N-dimensional array containing the regression for each input sequence (reverse).
+        """
+        X = X[:, 1:-1, :]
+        ry, y, fy = self.aer_model.predict(X)
+        return ry, y, fy
+
+
+def bi_regression_errors(y: ndarray, ry_hat: ndarray, fy_hat: ndarray,
+                         smoothing_window: float = 0.01, smooth: bool = True, mask: bool = True):
+    """Compute an array of absolute errors comparing the forward and reverse predictions with
+    the expected output.
+
+    Anomaly scores are created in the forward and reverse directions. Scores in overlapping indices
+    are averaged while scores in non-overlapping indices are taken directly from either forward or
+    reverse anomaly scores.
+
+    If smooth is True, apply EWMA to the resulting array of errors.
+
+    Args:
+        y (ndarray):
+            Ground truth.
+        ry_hat (ndarray):
+            Predicted values (reverse).
+        fy_hat (ndarray):
+            Predicted values (forward).
+        smoothing_window (float):
+            Optional. Size of the smoothing window, expressed as a proportion of the total
+            length of y. If not given, 0.01 is used.
+        smooth (bool):
+            Optional. Indicates whether the returned errors should be smoothed with EWMA.
+            If not given, `True` is used.
+        mask (bool): bool = True
+            Optional. Mask anomaly score errors in the beginning.
+            If not given, `True` is used.
+    Returns:
```

Comment:
minor: add space between the last a item in argument and returns.

Reply:
Added